### PR TITLE
fixed training bugs in siamese network example

### PR DESCRIPTION
**Pull request by hazemessamm**
There were some training and inference bugs in siamese network with triplet loss example, the bugs were the following:
1. In `train_step` we should pass `training=True`.
2. In the `test_step` we should pass `training=False`.
3. In inference we should pass each example to the `CosineSimilarity` metric because passing multiple example will compute the average and we want the similarity per example.
4. `EarlyStopping` should be used to get better accuracy.

### CODE REVIEW

File: examples/vision/siamese_network.py [hunk 1]
```diff
@@ -412,9 +434,4 @@ def metrics(self):
 [Intro to Keras for researchers](https://keras.io/getting_started/intro_to_keras_for_researchers/)
 and [Writing a training loop from scratch](https://www.tensorflow.org/guide/keras/writing_a_training_loop_from_scratch?hl=en).
 
-
-**Example available on HuggingFace**
```
Comment (fchollet) [inline examples/vision/siamese_network.py:416]: Please keep the link.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR. The changes look good to me. Please add the generated files (and keep the HF link)

**Review by fchollet — COMMENTED** [review]
LGTM, please add the generated files.

**Comment by pcoet** [comment]
Hi @hazemessamm, thanks for the PR. Could you add the generated files? Let us know if you're still working on this. Otherwise we'll close the request. Thanks!

**Comment by hazemessamm** [comment]
Yes sure, Sorry, I forgot that I did not submit the generated files, currently I do not have them but I will re-generate them and will submit it.
